File: src/main/java/com/football/booking/entity/Field.java
package com.football.booking.entity;

import com.football.booking.enums.FieldType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "fields")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Field {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String address;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private FieldType fieldType;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal pricePerHour;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "field", cascade = CascadeType.ALL)
    @Builder.Default
    private List<Booking> bookings = new ArrayList<>();
}

File: iOS/Ploshchadka/Models/Field.swift
import Foundation

struct Field: Codable, Identifiable {
    let id: Int64
    let name: String
    let address: String
    let fieldType: String
    let pricePerHour: Double
    let description: String?
    let isActive: Bool
    let ownerUsername: String?
    let createdAt: String?

    var isIndoor: Bool { fieldType == "INDOOR" }

    var fieldTypeLabel: String { isIndoor ? "Крытое" : "Открытое" }

    var formattedPrice: String { "\(Int(pricePerHour)) ₽/час" }
}
